pkg/river/encoding/encoding.go

```diff
@@ -0,0 +1,148 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/agent/pkg/river/token/builder"
+
+	"github.com/grafana/agent/pkg/river/internal/rivertags"
+	"github.com/grafana/agent/pkg/river/internal/value"
+)
+
+const attr = "attr"
+const object = "object"
```

Comment:
nit: decorate the names here to make it more clear how they're intended to be used, like `attrType` and `objectType`.